src/arrayMethodJoin.js

```diff
@@ -4,8 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const len = this.length;
+
+    for (let i = 0; i < len; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
```

Comment:
Replace with `this[i]`, it will be the same as `this[i] !== null && this[i] !== undefined`

Reply:
![image](https://user-images.githubusercontent.com/117356906/222886721-4565f0c6-2032-4fbc-ac11-82c1f5265d6f.png)
but we lose 0 NaN